### ISSUE TITLE
On Ubuntu 24.04.1 LTS, SuperProductivity doesn't get mimized to tray anymore

### ISSUE THREAD

**Issue by hkockerbeck** [johannesjo/super-productivity#3427]
### Your Environment

- Version used: Snap 10.0.2
- Operating System and version: Ubuntu 24.04.1 LTS
- Desktop Environment: Gnome 46.0
- Browser Name and version: N/A

### Expected Behavior

If I activate "Minimize to tray (desktop only)" in the settings and close the app by using the close window button, the app should be minimized to the system tray.

### Current Behavior

The app is closed completely. If I start the app from a terminal, I get the following error (among stuff about my GPU, which is probably unrelated):
```
TypeError: Cannot read properties of undefined (reading 'hide')
    at /snap/superproductivity/2036/resources/app.asar/electron/main-window.js:199:30
    at /snap/superproductivity/2036/resources/app.asar/electron/get-settings.js:17:44
    at Array.forEach (<anonymous>)
    at IpcMainImpl.getSettingsCb (/snap/superproductivity/2036/resources/app.asar/electron/get-settings.js:17:13)
```

Additionally, there's a message in the terminal during startup that may or may not be related (AppArmor not granting access to something, so it's `undefined` later? Just a wild guess, though):
```
[10807:0909/080527.397845:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.login1.Manager.Inhibit: object_path= /org/freedesktop/login1: org.freedesktop.DBus.Error.AccessDenied: An AppArmor policy prevents this sender from sending this message to this recipient; type="method_call", sender=":1.174" (uid=1000 pid=10807 comm="/snap/superproductivity/2036/superproductivity --n" label="snap.superproductivity.superproductivity (enforce)") interface="org.freedesktop.login1.Manager" member="Inhibit" error name="(unset)" requested_reply="0" destination="org.freedesktop.login1" (uid=0 pid=1312 comm="/usr/lib/systemd/systemd-logind" label="unconfined")
```

### Steps to Reproduce (for bugs)

1. Start the app with "Minimize to tray" activated in the settings.
2. Close the app by clicking on the close window button.
3. The app doesn't get minimized to tray, but is closed completely

### Can you reproduce this reliably?

I can reproduce this reliably.

### Console Output

No output in the console that appears to be related to this.

### Error Log (Desktop only)

The latest entries in main.log are from late July (which may be an indication for another problem?), so probably not relevant for this issue. This issue popped up quite recently, within the last few days.

**Comment by srv-stack** [comment]
I confirm the error:

System: 
`Distributor ID:	Ubuntu
Description:	Ubuntu 24.04.1 LTS
Release:	24.04
`
log:
`2024-09-09T10:11:53.718771+03:00 gabi superproductivity_superproductivity.desktop[30004]: [30004:0909/101153.718658:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.login1.Manager.Inhibit: object_path= /org/freedesktop/login1: org.freedesktop.DBus.Error.AccessDenied: An AppArmor policy prevents this sender from sending this message to this recipient; type="method_call", sender=":1.337" (uid=1000 pid=30004 comm="/snap/superproductivity/2036/superproductivity --n" label="snap.superproductivity.superproductivity (enforce)") interface="org.freedesktop.login1.Manager" member="Inhibit" error name="(unset)" requested_reply="0" destination="org.freedesktop.login1" (uid=0 pid=1392 comm="/usr/lib/systemd/systemd-logind" label="unconfined")
2024-09-09T10:11:54.574057+03:00 gabi superproductivity_superproductivity.desktop[30004]: 10:11:54.573 › No custom styles detected at /home/gabi/snap/superproductivity/common/.config/superProductivity/styles.css
2024-09-09T10:12:07.069401+03:00 gabi superproductivity_superproductivity.desktop[30004]: TypeError: Cannot read properties of undefined (reading 'show')
2024-09-09T10:12:07.069510+03:00 gabi superproductivity_superproductivity.desktop[30004]:     at /snap/superproductivity/2036/resources/app.asar/electron/main-window.js:232:30
2024-09-09T10:12:07.069551+03:00 gabi superproductivity_superproductivity.desktop[30004]:     at /snap/superproductivity/2036/resources/app.asar/electron/get-settings.js:17:44
2024-09-09T10:12:07.069579+03:00 gabi superproductivity_superproductivity.desktop[30004]:     at Array.forEach (<anonymous>)
2024-09-09T10:12:07.069606+03:00 gabi superproductivity_superproductivity.desktop[30004]:     at IpcMainImpl.getSettingsCb (/snap/superproductivity/2036/resources/app.asar/electron/get-settings.js:17:13)
2024-09-09T10:12:07.069644+03:00 gabi superproductivity_superproductivity.desktop[30004]:     at IpcMainImpl.emit (node:events:519:28)
2024-09-09T10:12:07.069672+03:00 gabi superproductivity_superproductivity.desktop[30004]:     at WebContents.<anonymous> (node:electron/js2c/browser_init:2:83159)
2024-09-09T10:12:07.069705+03:00 gabi superproductivity_superproductivity.desktop[30004]:     at WebContents.emit (node:events:519:28)
`

**Comment by KatrinIhler** [comment]
Same here!

**Comment by AlaricLightin** [comment]
I have the same behaviour on Ubuntu 22.04.4 LTS.

**Comment by hkockerbeck** [comment]
> I have the same behaviour on Ubuntu 22.04.4 LTS.

That may be of interest. I've used Ubuntu 22.04.x LTS until (if I remember correctly) about a week ago, and the problem didn't occur with the then-current versions of the app. So I would assume the cause to be changes in the app and not (only) any difference between the OS versions.

**Comment by jothanne** [comment]
The same here on Windows 10 Pro - and in addition the minimize button closes the app too.

**Comment by johannesjo** [comment]
I am working on it! Thanks for reporting!

**Comment by srv-stack** [comment]
> I am working on it! Thanks for reporting!

I confirm it's fixed in ubuntu 24.04.1 LTS with the 10.0.4 release

**Comment by johannesjo** [comment]
Thanks everybody! Closing this.
